tests/TransportTest.php

```diff
@@ -28,13 +28,34 @@
 use OpenSearch\Transport;
 use GuzzleHttp\Ring\Future\FutureArray;
 use GuzzleHttp\Ring\Future\FutureArrayInterface;
+use PHPUnit\Framework\MockObject\MockObject;
 use PHPUnit\Framework\TestCase;
 use Psr\Log\LoggerInterface;
 use React\Promise\Deferred;
-use React\Promise\Promise;
 
 class TransportTest extends TestCase
 {
+    /**
+     * @var Connection|MockObject
```

Comment:
Do we need this union of MockObject? Is Connection not enough?

Reply:
IMO yes. Because in test we use it as Connection and as MockObject in some context (when mock method behaviors)